internal/fluentbit/config/builder/kubernetes_filter.go

```diff
@@ -16,8 +16,11 @@ func createKubernetesFilter(pipeline *telemetryv1alpha1.LogPipeline) string {
 		AddConfigParam("kube_tag_prefix", fmt.Sprintf("%s.var.log.containers.", pipeline.Name)).
 		AddConfigParam("annotations", fmt.Sprintf("%v", fluentBitFlag(pipeline.Spec.Input.Application.KeepAnnotations))).
 		AddConfigParam("labels", fmt.Sprintf("%v", fluentBitFlag(!pipeline.Spec.Input.Application.DropLabels))).
-		AddConfigParam("buffer_size", "1MB").
-		Build()
+		AddConfigParam("buffer_size", "1MB")
+	if dropLogBody(pipeline) {
+		sb.AddConfigParam("keep_log", "off")
```

Comment:
Please render the param always and just switch the value, that will make it easier for debugging as you explicitly see the setting and you do not need to remember the fluenbit default. Also we want to switch our default and then you would need to render it anyway always.

Reply:
The filter configuration parameter is now always present